client/components/place/place-list/place-list.component.js

```diff
@@ -204,6 +202,8 @@ angular.module('placeList', ['filterMapType', 'popularTracks', 'ngAnimate'])
               placesOnMap.showPlaces(places);
               angular.element('#spinner').removeClass('spinner');
             });
+          } else {
+            angular.element('#spinner').removeClass('spinner');
```

Comment:
Why do you adding and then immediately removing spinner?)
You can just move line 185 inside `if (realPlaceType.length)`